lms/extractors/base.py

```diff
@@ -82,6 +84,14 @@ def __iter__(self) -> Iterator[Tuple[int, List[File], str]]:
             log.debug(f'Trying extractor: {cls.__name__}')
             extractor = cls(to_extract=self.to_extract)
             if extractor.can_extract():
+                if (
```

Comment:
We shouldn't assume anything about the subclasses of this class.

Maybe put it into the `get_exercises` of ZipArchive?